src/lakefs_spec/client_helpers.py

```diff
@@ -189,11 +185,35 @@ def create_tag(
     try:
         return client.tags_api.create_tag(repository=repository, tag_creation=tag_creation)
     except ApiException as e:
-        if e.status == 409 and exist_ok:
-            return client.tags_api.get_tag(repository=repository, tag=tag)
+        if e.status == 409:
+            target_commit = rev_parse(client=client, repository=repository, ref=ref)
+            existing_tag = client.tags_api.get_tag(repository=repository, tag=tag)
+            if existing_tag.commit_id == target_commit.id:
+                return existing_tag
         raise e
 
 
+def delete_tag(client: LakeFSClient, repository: str, tag: str | Ref) -> None:
+    """
+    Delete the specified tag from a repository.
+
+    Parameters
+    ----------
+    client : LakeFSClient
+        lakeFS client object.
+    repository : str
+        Name of the repository from which the tag will be deleted.
+    tag : str
+        Name of the tag to be deleted.
+    """
+    if isinstance(tag, str):
+        client.tags_api.delete_tag(repository=repository, tag=tag)
+    elif isinstance(tag, Ref):
+        client.tags_api.delete_tag(repository=repository, tag=tag.id)
+    else:
+        raise ValueError(r"Type of tag must be str or Ref.")
```

Comment:
This is unnecessary with a proper interface typing - after we tell users what is allowed, it's not our responsibility anymore if they pass something else.

I would go with this:

```suggestion
    if isinstance(tag, Ref):
        tag = tag.id

    client.tags_api.delete_tag(repository=repository, tag=tag.id)
```